packages/calcite-components/src/components/input-date-picker/input-date-picker.tsx

```diff
@@ -155,8 +157,16 @@ export class InputDatePicker
       let newValueAsDate: Date | Date[];
 
       if (Array.isArray(newValue)) {
+        if (isTwoDigitYear(newValue[0]) || isTwoDigitYear(newValue[1])) {
```

Comment:
Reason for having separate method `isTwoDigitYear` is to avoid multiple method invokations when we are mutating `value` prop.